sdk/core/Azure.Core/tests/ClientTestBaseTests.cs

```diff
@@ -226,6 +261,81 @@ public virtual TestClient GetAnotherTestClient()
                 return new TestClient();
             }
             public virtual Operations SubProperty => new Operations();
+
+            // For checking DaignsoticScope concurrency issues.
+            private readonly Random _random = new Random();
+
+            public virtual string MethodA()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodA)}");
+                scope.Start();
+
+                return nameof(MethodA);
+            }
+
+            public virtual async Task<string> MethodAAsync()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodA)}");
+                scope.Start();
+
+                await Task.Delay(_random.Next(10, 50));
+                return nameof(MethodAAsync);
+            }
+
+            public virtual string MethodB()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodB)}");
+                scope.Start();
+
+                return nameof(MethodB);
+            }
+
+            public virtual async Task<string> MethodBAsync()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodB)}");
+                scope.Start();
+
+                await Task.Delay(_random.Next(10, 50));
+                return nameof(MethodAAsync);
+            }
+
+            public virtual string MethodC()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodC)}");
+                scope.Start();
+
+                return nameof(MethodC);
+            }
+
+            public virtual async Task<string> MethodCAsync()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodC)}");
+                scope.Start();
+
+                await Task.Delay(_random.Next(10, 50));
+                return nameof(MethodAAsync);
+            }
+
+            public virtual string MethodD()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodD)}");
+                scope.Start();
+
+                return nameof(MethodD);
+            }
+
+            public virtual async Task<string> MethodDAsync()
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TestClient)}.{nameof(MethodD)}");
+                scope.Start();
+
+                await Task.Delay(_random.Next(10, 50));
```

Comment:
I know, but in this case it doesn't really need to be. It'll still provide enough randomness for testing. I wouldn't ship this in production, but locking just to get some "noise" would mitigate some amount of noise to synchronize.

Reply:
That's not the way random is thread-unsafe. It locks up when used from multiple threads and starts returning zeroes.

Reply:
Good to know. I've never seen that behavior, but I don't often use `Random` anyway. I'll keep that mind. Rssolving this conversation, though, since you just used `Task.Yield`. I had that in a previous iteration but expected it was still too fast and switched to delays since I couldn't repro the original issue on my dev box.